freqtrade/exchange/__init__.py

```diff
@@ -105,132 +94,231 @@ def validate_pairs(pairs: List[str]) -> None:
     if 'fetchOHLCV' not in _API.has or not _API.has['fetchOHLCV']:
-        logger.warning('Exhange %s does not support fetching historical candlestick data.',
-                       _API.name)
-        return []
+        raise OperationalException(
+            'Exhange {} does not support fetching historical candlestick data.'.format(_API.name)
+        )
 
     try:
-        ohlcv = _API.fetch_ohlcv(pair, timeframe=str(tick_interval)+"m")
-        return ohlcv
+        history = _API.fetch_ohlcv(pair, timeframe=tick_interval)
+        history_json = []
+        for candlestick in history:
+            history_json.append({
+                'T': arrow.get(candlestick[0]/1000.0).strftime('%Y-%m-%dT%H:%M:%S.%f'),
+                'O': candlestick[1],
+                'H': candlestick[2],
+                'L': candlestick[3],
+                'C': candlestick[4],
+                'V': candlestick[5],
+            })
+        return history_json
     except IndexError as e:
         logger.warning('Empty ticker history. Msg %s', str(e))
+        return []
     except ccxt.NetworkError as e:
-        logger.warning('Could not load ticker history due to networking error. Message: %s', str(e))
+        raise NetworkException(
+            'Could not load ticker history due to networking error. Message: {}'.format(e)
+        )
     except ccxt.BaseError as e:
-        logger.warning('Could not fetch ticker data. Msg: %s', str(e))
-    return []
+        raise OperationalException('Could not fetch ticker data. Msg: {}'.format(e))
 
 
-def cancel_order(order_id: str) -> None:
+def cancel_order(order_id: str, pair: str) -> None:
     if _CONF['dry_run']:
         return
 
-    return _API.cancel_order(order_id)
+    try:
+        return _API.cancel_order(order_id, pair)
+    except ccxt.NetworkError as e:
+        raise NetworkException(
+            'Could not get order due to networking error. Message: {}'.format(e)
+        )
+    except ccxt.InvalidOrder as e:
+        raise DependencyException(
+            'Could not cancel order. Message: {}'.format(e)
+        )
+    except ccxt.BaseError as e:
+        raise OperationalException(e)
 
 
-def get_order(order_id: str) -> Dict:
+def get_order(order_id: str, pair: str) -> Dict:
     if _CONF['dry_run']:
         order = _DRY_RUN_OPEN_ORDERS[order_id]
         order.update({
             'id': order_id
         })
         return order
-
-    return _API.get_order(order_id)
+    try:
+        return _API.fetch_order(order_id, pair)
+    except ccxt.NetworkError as e:
+        raise NetworkException(
+            'Could not get order due to networking error. Message: {}'.format(e)
+        )
+    except ccxt.InvalidOrder as e:
+        raise DependencyException(
+            'Could not get order. Message: {}'.format(e)
+        )
+    except ccxt.BaseError as e:
+        raise OperationalException(e)
 
 
+# TODO: reimplement, not part of ccxt
```

Comment:
getting the exchange url for the pair is there already.

Reply:
It was a bit half-done, but I fixed it up!